### DISCUSSION TITLE
Module as a component library with component prefix and auto import

### DISCUSSION THREAD

**Discussion by ianjamieson** [nuxt/nuxt#21298] [Questions]
I have a Stackblitz example here, https://stackblitz.com/github/ianjamieson/nuxt-test-module?file=README.md, which I created using:

`npx nuxi init -t module nuxt-test-module`

The issue I am having is around the component prefix, you'll notice that in the playground (essentially the consuming app) I have this config:

```
export default defineNuxtConfig({
  modules: ['../src/module'],
  myModule: {},
  components: {
    dirs: [
      '~/components',
      {
        path: '../src/components',
        prefix: 'PREFIX-',
      },
    ],
  },
})
```

So any components that are registered as part of the module will need to have the `PREFIX-`.

This works to some extent, as you can see in `playground/app.vue`:

```
<template>
  <div>
    <LocalComp />
    <PREFIX-ModuleComp />
  </div>
</template>

<script setup>
</script>
```

I can use the `<PREFIX-ModuleComp />` component and it renders as expected. However, within that component I am trying to call another component from within the module, so `ModuleComp.vue` looks like this:

```
<template>
  This is a module component
  <AnotherModuleComp />
</template>
```

Now `<AnotherModuleComp />` does not exist. I can put the prefix on it like `<PREFIX-AnotherModuleComp />` and that does work. But it is not the functionality I want. 

I want the module (which is a component library) to be able to use all its own components without a prefix and then if any projects use it then they should be able to set the prefix and use the components with whatever prefix that project defines.

**Comment by maou-shonen** [comment]
In the module, you should forget about auto-imports.
